Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -327,14 +334,14 @@ static void CheckCPConfiguration(int vtx_attr_group)
 template <bool IsPreprocess>
 int RunVertices(int vtx_attr_group, OpcodeDecoder::Primitive primitive, int count, DataReader src)
 {
-  if (count == 0)
+  if (count == 0) [[unlikely]]
     return 0;
   ASSERT(count > 0);
 
   VertexLoaderBase* loader = RefreshLoader<IsPreprocess>(vtx_attr_group);
 
   int size = count * loader->m_vertex_size;
-  if ((int)src.size() < size)
+  if ((int)src.size() < size) [[unlikely]]
     return -1;
```

Comment:
This check can probably be removed entirely, actually, as it's redundant with this:

https://github.com/dolphin-emu/dolphin/blob/c8705f014d1026b10915938ce3d798255ca56f20/Source/Core/VideoCommon/OpcodeDecoding.h#L234-L240

and in fact this return value isn't properly handled anymore:

https://github.com/dolphin-emu/dolphin/blob/c8705f014d1026b10915938ce3d798255ca56f20/Source/Core/VideoCommon/OpcodeDecoding.cpp#L122-L127

It doesn't need to be removed in this PR, but this is probably something that can be refactored later (along with the DataReader itself since it isn't really doing much anymore to my understanding).

Reply:
I think I'll leave that to you since you understand that code muuuuch better than me. ;)